assets/src/edit-story/app/api/useAPI.js

```diff
@@ -14,18 +14,14 @@
  * limitations under the License.
  */
 
-/**
- * External dependencies
- */
-import { useContext } from 'react';
-
 /**
  * Internal dependencies
  */
+import { identity, useContextSelectorShallow } from '../../utils/context';
 import Context from './context';
 
-function useAPI() {
-  return useContext(Context);
+function useAPI(selector) {
+  return useContextSelectorShallow(Context, selector ?? identity);
```

Comment:
Hmm. This is an interesting twist. Would we allow a selector function to change ever? Does the framework make it changeable or is only the first one used? Otherwise, I feel like the `selector ?? identity` will be used a lot, so I think we need to push it back to the implementation of the `useContextSelectorShallow`. As an optimization, when `identity` is used, it also means you actually don't need to do any shallow comparison. I don't know what the library we have does. Perhaps they already do `if (a === b) return true`, but I think being explicit would be helpful.

Reply:
The selector function can change for any call to useContextSelector, so two calls to useContextSelector can select different fragments.

In the final changes, 'selector' will be a mandatory argument for useMedia, useStory and useCanvas where performance is more critical, but optional for useAPI, useConfig etc where changes happen infrequently (or not at all).

Reply:
Sounds good.